### PR TITLE
Update alteryx_open_src_update_checker to 2.0.0

### PR DESCRIPTION
**Pull request by gsheni**

### CODE REVIEW

**Review by frances-h — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 extras_require = {
     'tsfresh': ['featuretools-tsfresh-primitives >= 0.1.0'],
-    'update_checker': ['featuretools-update-checker >= 1.0.0'],
+    'update_checker': ['featuretools-update-checker >= 2.0.0'],
```
Comment (frances-h) [inline setup.py]: should be `'update_checker': ['alteryx-open-src-update-checker >= 2.0.0']`

**Review by frances-h — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (gsheni) [reply setup.py]: Fixed

**Review by frances-h — DISMISSED** [review]

File: featuretools/__init__.py [hunk 2]
```diff
@@ -35,11 +35,11 @@
 logger = logging.getLogger('featuretools')
 
 # Call functions registered by other libraries when featuretools is imported
-for entry_point in pkg_resources.iter_entry_points('featuretools_initialize'):
+for entry_point in pkg_resources.iter_entry_points('alteryx_open_src_initialize'):
     try:
         method = entry_point.load()
         if callable(method):
-            method()
+            method('featuretools')
```
Comment (rwedge) [inline featuretools/__init__.py:42]: won't this break any existing plugins that use the `featuretools_intialize` entry point?

Comment (gsheni) [reply featuretools/__init__.py:42]: - There isn't any existing plugins (that we can search for) besides the update checker that uses that entry point:
  - https://github.com/search?q=featuretools_intialize&type=code

Comment (gsheni) [reply featuretools/__init__.py:42]: I could make this change be in another for loop...

Comment (rwedge) [reply featuretools/__init__.py:42]: add another loop for now and float an issue on whether to remove the old entry point?

Comment (gsheni) [reply featuretools/__init__.py]: Sounds good. Fixed the MR and issue created: https://github.com/alteryx/featuretools/issues/1461

**Review by rwedge — APPROVED** [review]
Looks good

**Pull request merged**